build/postgres-operator/Dockerfile

```diff
@@ -19,7 +19,7 @@ RUN mkdir -p build/_output/bin \
        go build -ldflags "-w -s -X main.GitCommit=$GIT_COMMIT -X main.GitBranch=$GIT_BRANCH" \
            -o build/_output/bin/postgres-operator \
                ./cmd/postgres-operator \
-    && CGO_ENABLED=$CGO_ENABLED GOOS=$GOOS GO_LDFLAGS=$GO_LDFLAGS \
+    && CGO_ENABLED=$CGO_ENABLED GOARCH=${TARGETARCH} GOOS=$GOOS GO_LDFLAGS=$GO_LDFLAGS \
```

Comment:
why do we need it?

Reply:
I had a following error in `custom-extensions` test:
```
$ kubectl logs custom-extensions-instance1-g95j-0 extension-installer
+ IFS=,
+ read -ra extensions
+ args=('-type' 's3' '-region' 'eu-central-1' '-bucket' 'pg-extensions' '-extension-path' '/pgdata/extension/16')
+ declare -a args
+ [[ -n '' ]]
+ for key in "${extensions[@]}"
+ '[' -f /pgdata/extension/16/pg_cron-pg16-1.6.1.installed ']'
+ echo 'Installing extension: pg_cron-pg16-1.6.1'
Installing extension: pg_cron-pg16-1.6.1
+ /usr/local/bin/extension-installer -type s3 -region eu-central-1 -bucket pg-extensions -extension-path /pgdata/extension/16 -key pg_cron-pg16-1.6.1 -install
/usr/local/bin/install-extensions.sh: line 27: /usr/local/bin/extension-installer: cannot execute binary file: Exec format error
```
Specifying an architecture fixes it